libeos-updater-util/avahi-service-file.c

```diff
@@ -257,3 +326,426 @@ eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+                         variant_to_binary_variant (repository_index_variant));
+  txt_records = g_variant_ref_sink (g_variant_builder_end (&records_builder));
+
+  return generate_avahi_service_template_to_file (service_file,
+                                                  "EOS OSTree update service on %h",
+                                                  "_repo._ostree._tcp",
+                                                  port,
+                                                  txt_records,
+                                                  cancellable,
+                                                  error);
+}
+
+static gboolean
+generate_ostree_avahi_v1_service_file (const gchar   *avahi_service_directory,
+                                       OstreeRepo    *repo,
+                                       GVariantDict  *options_dict,
+                                       GCancellable  *cancellable,
+                                       GError       **error)
+{
+  g_autoptr(GVariant) version_variant = NULL;
+  g_autoptr(GVariant) refs_bloom_filter_variant = NULL;
+  g_autoptr(GVariant) summary_timestamp_variant = NULL;
+  g_autoptr(GVariant) repository_index_variant = NULL;
+  g_autoptr(GFile) service_file = NULL;
+  guint16 repository_index = get_repository_index (options_dict);
+  guint16 port = get_avahi_service_port (options_dict);
+  guint64 summary_timestamp;
+
+  version_variant = variant_sink_new ("y", 1);
+  g_assert (g_variant_type_equal (g_variant_get_type (version_variant),
+                                  EOS_OSTREE_AVAHI_VERSION_VARIANT_TYPE));
+  refs_bloom_filter_variant = get_bloom_filter_variant (repo,
+                                                        options_dict,
+                                                        cancellable,
+                                                        error);
+  if (refs_bloom_filter_variant == NULL)
+    {
+      return FALSE;
+    }
+  g_assert (g_variant_type_equal (g_variant_get_type (refs_bloom_filter_variant),
+                                  EOS_OSTREE_AVAHI_V1_REFS_BLOOM_FILTER_VARIANT_TYPE));
+  if (!get_summary_timestamp (repo, cancellable, &summary_timestamp, error))
+    {
+      return FALSE;
+    }
+  summary_timestamp_variant = variant_sink_new ("t", summary_timestamp);
+  g_assert (g_variant_type_equal (g_variant_get_type (summary_timestamp_variant),
+                                  EOS_OSTREE_AVAHI_V1_SUMMARY_TIMESTAMP_VARIANT_TYPE));
+  repository_index_variant = variant_sink_new ("q", repository_index);
+  g_assert (g_variant_type_equal (g_variant_get_type (repository_index_variant),
+                                  EOS_OSTREE_AVAHI_V1_REPOSITORY_INDEX_VARIANT_TYPE));
+
+  service_file = get_ostree_service_file (avahi_service_directory,
+                                          repository_index);
+
+  return generate_ostree_avahi_v1_service_file_from_variants (service_file,
```

Comment:
As stated above - I'm not sure that the function that does not store the gvariant and does not modify it should ever sink or ref it.